store-failures

```diff
@@ -55,46 +55,79 @@ def main():
+        context = status["context"]
+        if "@" in context:
+            parts = context.split("@")
             context = parts[0]
             if not parts[1].startswith("bots"):
                 # Can contain branch, as it diffs between master and branches (e.g `cockpit-project/cockpit/rhel-7.9`)
                 repo = parts[1]
 
-        row = cursor.execute("SELECT id FROM TestRuns WHERE url = ?", (url, )).fetchone()
-        if row is not None:
-            logging.warning("Test run %s already exists as ID %i, skipping", url, row[0])
-            continue
-
-        raw_log = url[:-5] if url.endswith(".html") else url
-        logging.debug("scanning log %s", raw_log)
-
-        # Parse time from log file name. Wrap in try/except block if the url would be unexpected
-        try:
-            parts = url.split("/")[-2].split("-")
-            day = parts[2]
-            hours = parts[3]
-            dtime = datetime.strptime("{0} {1}".format(day, hours), "%Y%m%d %H%M%S")
-            time = datetime.timestamp(dtime)
-        except IndexError:
-            time = datetime.timestamp(datetime.now())
-
-        with urllib.request.urlopen(raw_log, context=ctx if "logs.cockpit-project.org" in raw_log else None) as fp:
-            for (testname, retry_reason) in find_failed_tests(fp):
-                if run_id is None:
-                    run_id = insert_run(cursor, repo, opts.revision, context, url, time)
-                insert_failure(cursor, run_id, testname, retry_reason)
+        if context not in contexts:
+            contexts[context] = []
+
+        if status["state"] == "pending":
+            if "Testing in progress" in status["description"]:
+                contexts[context][-1]["url"] = status["target_url"]
+                contexts[context][-1]["started"] = datetime.strptime(status["created_at"], "%Y-%m-%dT%H:%M:%S%z").timestamp()
+            elif "Not yet tested" in status["description"]:
+                new_run = {"posted": datetime.strptime(status["created_at"], "%Y-%m-%dT%H:%M:%S%z").timestamp(), "repo": repo}
+                contexts[context].append(new_run)
+            else:
+                logging.warning("Pending status has unexpected description '{0}', skipping.".format(status["description"]))
+        elif status["state"] in ["success", "failure", "error"]:
+            contexts[context][-1]["finished"] = datetime.strptime(status["created_at"], "%Y-%m-%dT%H:%M:%S%z").timestamp()
+            contexts[context][-1]["state"] = status["state"]
+            if status["state"] == "error":
+                contexts[context][-1]["description"] = status["description"]
+        else:
+            logging.warning("Status has unexpected state '{0}', skipping.".format(status["state"]))
+
+    for context, statuses in contexts.items():
+        for retry, status in enumerate(statuses):
+            if not status.get("finished"):
+                # TODO: Store partial, store as non-finished?
```

Comment:
Indeed. This would point to those 'Never picked up, we need to requeue it manually' cases, which surely contribute to us exceeding error budget by needing to manually intervene. It goes towards "All PRs finish up in 90 minutes" (or how we defined it).
But I want to do this as follow up, as it has a bit different logic then rest of finished tests.